e2e/app/page/notebook-cell.ts

```diff
@@ -0,0 +1,177 @@
+import {ElementHandle, Frame, Page} from 'puppeteer';
+import {getPropValue} from 'utils/element-utils';
+
+export enum CellType {
+  // To append to css selector
+  Code = '.code_cell',
+  Markdown = '.text_cell',
+  Any = '',
+}
+
+/**
+ * Notebook Cell represents the root element that contains both the code input and output cells.
+ */
+export default class NotebookCell {
+
+  private readonly page: Page;
+  private iframe: Frame; // Jupyter notebook iframe
+  private cellIndex: number; // 1-based cell index because notebook cell prompt index starts from 1.
```

Comment:
To be clear, the numbers show in the Jupyter UI are not cell indices. They correspond to the execution history of the notebook. For example, you can execute the 4th cell before you execute the first one. If you do this, you're notebook will look like:

```
[2] cell0
    cell1
    cell2
[1] cell3
```

When you start the notebook, no cell will have this label. If you always execute the notebook in a linear fashion, then they will look like they are just 1-indexed cell labels, so I can see how this could be confusing to someone who is debugging the test.

Personally, I would still prefer 0-indexed here. This is probably what I would assume the test code would do, rather than yielding to the UI.

Reply:
If I understand your argument, you're saying that you're making it easy for non-coders by using 1-based indexing which they are probably more familiar with than 0-indexing. If so, this is just doing a disservice to them: just use standard coding practices so they can learn the standards. The nth-child CSS selector is an implementation detail that the test writer does not need to concern themselves with.

So, I'm not convinced. But also I don't feel strongly enough to block the review. Fine to merge for now, happy to discuss more if you are not convinced whether this is worth changing in a follow-up.